src/IssueViz/Editor/IIssueSpanCalculator.cs

```diff
@@ -97,21 +89,28 @@ internal IssueSpanCalculator(IChecksumCalculator checksumCalculator)
 
             var start = new SnapshotPoint(currentSnapshot, startPos);
             var end = new SnapshotPoint(currentSnapshot, endPos);
+            var snapshotSpan = new SnapshotSpan(start, end);
+
+            if (IsHashDifferent(range.LineHash, snapshotSpan.GetText()) &&
+                IsHashDifferent(range.LineHash, startLine.GetText()))
+            {
+                // Out of sync: the range reported in the diagnostic has been edited, so we can no longer calculate the span
+                return EmptySpan;
+            }
 
-            return new SnapshotSpan(start, end);
+            return snapshotSpan;
         }
 
-        private bool IsLineHashDifferent(ITextRange range, ITextSnapshotLine snapshotLine)
+        private bool IsHashDifferent(string issueHash, string documentText)
         {
-            if (string.IsNullOrEmpty(range.LineHash))
+            if (string.IsNullOrEmpty(issueHash))
             {
                 return false;
             }
 
-            var textInSnapshot = snapshotLine.GetText();
-            var snapshotLineHash = checksumCalculator.Calculate(textInSnapshot);
+            var documentTextHash = checksumCalculator.Calculate(documentText);
 
-            return snapshotLineHash != range.LineHash;
+            return documentTextHash != issueHash;
```

Comment:
Just a question: why is line hash a string and not a number? Looks so weird

Reply:
No particular reason. The SQ APIs return the hashes as strings and we didn't change the format when we deserialized them.